src/main/java/org/fundacionjala/coding/melvi/BankOcr.java

```diff
@@ -0,0 +1,266 @@
+ * You work for a bank, which has recently purchased an ingenious machine to assist
+ * in reading letters and faxes sent in by branch offices. The machine scans the paper
+ * documents, and produces a file with a number of entries which each look like this:
+ * <p>
+ * _  _     _  _  _  _  _
+ * | _| _||_||_ |_   ||_||_|
+ * ||_  _|  | _||_|  ||_| _|
+ * Each entry is 4 lines long, and each line has 27 characters. The first 3 lines of
+ * each entry contain an account number written using pipes and underscores, and the
+ * fourth line is blank. Each account number should have 9 digits, all of which should
+ * be in the range 0-9. A normal file contains around 500 entries.
+ * <p>
+ * Your first task is to write a program that can take this file and parse it into actual
+ * account numbers.
+ * <p>
+ * User Story 2
+ * Having done that, you quickly realize that the ingenious machine is not in fact
+ * infallible. Sometimes it goes wrong in its scanning. The next step therefore is
+ * to validate that the numbers you read are in fact valid account numbers. A valid
+ * account number has a valid checksum. This can be calculated as follows:
+ * <p>
+ * account number:  3  4  5  8  8  2  8  6  5
+ * position names:  d9 d8 d7 d6 d5 d4 d3 d2 d1
+ * <p>
+ * checksum calculation:
+ * (d1+2*d2+3*d3+...+9*d9) mod 11 = 0
+ * So now you should also write some code that calculates the checksum for a given
+ * number, and identifies if it is a valid account number.
+ * <p>
+ * User Story 3
+ * Your boss is keen to see your results. He asks you to write out a file of your
+ * findings, one for each input file, in this format:
+ * <p>
+ * 457508000
+ * 664371495 ERR
+ * 86110??36 ILL
+ * ie the file has one account number per row. If some characters are illegible,
+ * they are replaced by a ?. In the case of a wrong checksum, or illegible number,
+ * this is noted in a second column indicating status.
+ */
+public class BankOcr {
+    static final String PATTERN0 = " _ | ||_|";
+    static final String PATTERN1 = "     |  |";
+    static final String PATTERN2 = " _  _||_ ";
+    static final String PATTERN3 = " _  _| _|";
+    static final String PATTERN4 = "   |_|  |";
+    static final String PATTERN5 = " _ |_  _|";
+    static final String PATTERN6 = " _ |_ |_|";
+    static final String PATTERN7 = " _   |  |";
+    static final String PATTERN8 = " _ |_||_|";
+    static final String PATTERN9 = " _ |_| _|";
+    public static final int TRHEE = 3;
+    public static final int INT = 3;
+    public static final int SIX = 6;
+    public static final int NINE = 9;
+    public static final int TWELVE = 12;
+    public static final int FIFTEEN = 15;
+    public static final int EIGHTEEN = 18;
+    public static final int TWENTY_ONE = 21;
+    public static final int TWENTY_FOUR = 24;
+    public static final int TWENTY_SEVEN = 27;
+    public static final int ONCE = 11;
+
+    private Map<String, String> charMap = new HashMap<String, String>() {
+        {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the contents of this initializer to a standard constructor or to field initializers. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1171)